src/content/guides/farmlands-pos-integration.mdoc

```diff
@@ -34,24 +34,27 @@ sequenceDiagram
 
 	participant Cardholder
 	participant POS
+	participant POS Server
 	participant Centrapay
 
 	Note over Cardholder: Present Barcode
 	Cardholder->>POS: Scan Barcode
 
-	POS->>Centrapay: Create Payment Request with Barcode
+  POS->>POS Server: Request Payment
 
-	loop Poll for Payment Confirmation
-		POS->>Centrapay: Get Payment Request
+	POS Server->>Centrapay: Create Payment Request with Barcode
 
-		alt Payment Condition Pending
+	loop Poll until final state
+		POS Server->>Centrapay: Get Payment Request
+
+		opt Payment Condition Pending
 			note over POS: ❓ Prompt Cashier to Accept or Decline Condition
-			POS->>Centrapay: POS Accepts Condition
-		else Payment Request Status is paid
-			Note over POS: Stop Polling for Confirmation
+			POS Server->>Centrapay: Accept Condition
```

Comment:
Change: This is missing an arrow where the cashier would accept the condition on the POS, then it goes to the POS server.